python_modules/dagster/dagster/_core/definitions/asset_check_evalaution.py

```diff
@@ -0,0 +1,78 @@
+from typing import Mapping, NamedTuple, Optional
+
+import dagster._check as check
+from dagster._core.definitions.events import AssetKey, MetadataValue
+from dagster._serdes import whitelist_for_serdes
+
+
+@whitelist_for_serdes
+class AssetCheckEvaluationPlanned(
+    NamedTuple(
+        "_AssetCheckEvaluationPlanned",
+        [
+            ("asset_key", AssetKey),
+            ("check_name", str),
+        ],
+    )
+):
+    def __new__(cls, asset_key: AssetKey, check_name: str):
+        return super(AssetCheckEvaluationPlanned, cls).__new__(
+            cls,
+            asset_key=check.inst_param(asset_key, "asset_key", AssetKey),
+            check_name=check.str_param(check_name, "check_name"),
+        )
+
+
+@whitelist_for_serdes
+class AssetCheckEvaluationTargetMaterialization(
```

Comment:
Wordy, but I think it would be more clear to call this `AssetCheckEvaluationTargetMaterializationData`. Could potentially leave out `Target`.

Reply:
Also needs docstring

Reply:
i like leaving target bc it clarifies the relationship to the materialization